### ISSUE TITLE
Opening a pandas (version 2.1.0) dataframe in Spyder shows a TypeError.

### ISSUE THREAD

**Issue by jpbusso** [spyder-ide/spyder#21350]
Problem Description

Opening a pandas (version 2.1.0) dataframe in Spyder shows an internal error.
What steps reproduce the problem?

    Install Spyder in Python environment with pandas 2.1.0
    Create a dataframe
    Open the dataframe in the variable explorer

What is the expected output? What do you see instead?

An internal error displays.
Paste Traceback/Error Below (if applicable)
Exception in comms call get_value:

  File "C:\Users\buj2870_admin\.conda\envs\benchmarking_table\Lib\site-packages\spyder_kernels\comms\commbase.py", line 347, in _handle_remote_call
    self._set_call_return_value(msg_dict, return_value)

  File "C:\Users\buj2870_admin\.conda\envs\benchmarking_table\Lib\site-packages\spyder_kernels\comms\commbase.py", line 384, in _set_call_return_value
    self._send_message('remote_call_reply', content=content, data=data,

  File "C:\Users\buj2870_admin\.conda\envs\benchmarking_table\Lib\site-packages\spyder_kernels\comms\frontendcomm.py", line 109, in _send_message
    return super(FrontendComm, self)._send_message(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "C:\Users\buj2870_admin\.conda\envs\benchmarking_table\Lib\site-packages\spyder_kernels\comms\commbase.py", line 247, in _send_message
    buffers = [cloudpickle.dumps(
               ^^^^^^^^^^^^^^^^^^

  File "C:\Users\buj2870_admin\.conda\envs\benchmarking_table\Lib\site-packages\cloudpickle\cloudpickle_fast.py", line 73, in dumps
    cp.dump(obj)

  File "C:\Users\buj2870_admin\.conda\envs\benchmarking_table\Lib\site-packages\cloudpickle\cloudpickle_fast.py", line 632, in dump
    return Pickler.dump(self, obj)
           ^^^^^^^^^^^^^^^^^^^^^^^

  File "C:\Users\buj2870_admin\.conda\envs\benchmarking_table\Lib\site-packages\pandas\core\arrays\arrow\array.py", line 615, in __getstate__
    state = self.__dict__.copy()
            ^^^^^^^^^^^^^

  File "C:\Users\buj2870_admin\.conda\envs\benchmarking_table\Lib\site-packages\pandas\core\arrays\string_arrow.py", line 461, in __getattribute__
    return partial(getattr(ArrowStringArrayMixin, item), self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

TypeError: the first argument must be callable

**Comment by jpbusso** [comment]
I think this might be because now pandas is integrating pyarrow types:

competition_name    string[pyarrow]
competition_id       int64[pyarrow]

Would it be possible for spyder to support showing these types as well?

**Comment by jitseniesen** [comment]
Thanks for your report. I did a quick test with Spyder 5.4.5 and Pandas 2.1.0, but it worked fine for me. However, I did not use PyArrow and I have never used PyArrow. Let's assume for the moment that the problem indeed is caused by pyarrow.

Could you please give us some instructions on how to reproduce the issue, bearing in mind that I don't know pyarrow? How can we create a dataframe that gives your error? Do you have pyarrow installed? What are your versions of Spyder and pandas, and any other package that may be relevant?

**Comment by jitseniesen** [comment]
I tried to reproduce this by installing Spyder in an environment with Pandas 2.1.2 and PyArrow 13.0.0. In case it is relevant, this is with Spyder master and cloudpickle 2.2.1. I created data frames with dtype `string[pyarrow_numpy]`, `string[pyarrow]`, `float[pyarrow]` and `int64[pyarrow]`. All these opened fine in the Variable Explorer. Since no further information was given on how to reproduce the bug, I will close this report.
